keps/0017-pipe-tasks.md

```diff
@@ -0,0 +1,122 @@
+---
+kep-number: 17
+title: Pipe Tasks
+authors:
+  - "@zen-dog"
+owners:
+  - "@zen-dog"
+editor: TBD
+creation-date: 2019-09-12
+last-updated: 2019-09-12
+status: provisional
+---
+
+# Pipe Tasks
+
+## Table of Contents
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+    * [Limitations](#limitations)
+    * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints-optional)
+* [Alternatives](#alternatives)
+
+## Summary
+
+Developing complicated operators often require generating files in one task and reusing them in a later one. A common example is generating custom certificates/dynamic configuration files in the bootstrap task and using them in the later deployment step of the service. This kep describes how KUDO could help operator developers by making this task seamless and automated.
+
+## Motivation
+
+We want to ease the development of operators with complicated life-cycles by enabling to pipe files from one task to another. While this can be achieved in a manual way, involving custom scripts, this is exactly the kind of "magic glue" that KUDO could provide to the operator developers to ease the development process.
+
+### Goals
+
+The goal of this kep is to describe application examples, limitations and ways to implement passing files between tasks. We aim to help operator developers pipe small files (<1Mb) like certificates and dynamically generated configuration which can be referenced later as ConfigMaps and Secrets.
```

Comment:
kep -> KEP :)

Reply:
I king of like goals a non-goals as a list of statements, it's easier to parse. So e.g. solving this for >1MB is a non-goal